hang-fds.go

```diff
@@ -39,11 +40,11 @@ func fdRaise(nn int) error {
 		return err
 	}
 
-	if rLimit.Cur >= n {
+	if uint64(rLimit.Cur) >= n {
 		fmt.Printf("already at %d >= %d fds\n", rLimit.Cur, n)
 		return nil // all good.
 	}
-	rLimit.Cur = n
+	*(*uint64)(unsafe.Pointer(&rLimit.Cur)) = n
```

Comment:
mmm... this sketches me out a little. Maybe an equivalent way (without using unsafe) could be:
var i interface{} = &rLimit.Cur
switch i := i.(type) {
case *uint64:
  *i = uint64(n)
case *int64:
  *i = int64(n)
default:
  return fmt.Errorf("error message")
}
```